src/main/java/net/mcreator/ui/dialogs/workspace/GeneratorSelector.java

```diff
@@ -29,22 +29,35 @@
 import net.mcreator.ui.component.util.PanelUtils;
 import net.mcreator.ui.init.L10N;
 import net.mcreator.ui.init.UIRES;
+import net.mcreator.workspace.ShareableZIPManager;
+import net.mcreator.workspace.Workspace;
 
 import javax.annotation.Nullable;
 import javax.swing.*;
 import javax.swing.plaf.basic.BasicProgressBarUI;
 import java.awt.*;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
+import java.io.File;
 import java.text.DecimalFormat;
 import java.util.Map;
 
 public class GeneratorSelector {
 
 	private static final String covpfx = "dialog.generator_selector.coverage.";
 
+	/**
+	 * <p>Open a dialog window to select a {@link Generator} from the loaded generators. </p>
+	 *
+	 * @param parent <p>The window to attach the dialog</p>
+	 * @param current <p>The current generator settings used</p>
+	 * @param currentFlavor <p>This is the current type of generator to use for the generator list.</p>
+	 * @param newWorkspace <p>Indicate if we need to use features specific to existing workspaces or not</p>
+	 * @param workspace <p>The workspace to backup</p>
+	 * @return <p>The {@link GeneratorConfiguration} to use</p>
+	 */
 	public static GeneratorConfiguration getGeneratorSelector(Window parent, @Nullable GeneratorConfiguration current,

```

Comment:
newWorkspace is then not needed, as when workspace is null, it is newWorkspace, otherwise, it is existing workspace.

Also workspace arg should be marked as nullable and handled as such